### PR TITLE
added ride validation

### PR DESCRIPTION
**Pull request by galpinto1308**
gas money >= 0
seats natural number and > 0 (There is no point in seats = 0)

### CODE REVIEW

File: apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx [hunk 1]
```diff
@@ -168,14 +188,17 @@ const CreateRideDialog = ({
...
       </DialogContent>
       <DialogActions>
         <Button onClick={handleClose} disabled={isLoading}>
           {TEXT.CANCEL}
         </Button>
-        <SubmitButton type="submit" disabled={isLoading} onClick={handleSubmit}>
+        <SubmitButton type="submit" disabled={isLoading || isGasMoneyNegative || isSeatsLessThanOne || isSeatsNotInteger} onClick={handleSubmit}>
           {isLoading ? (
             <CircularProgress size={24} color="info" />
           ) : (
```
Comment (Zoeshwa) [inline apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx:204]: what about update ride?

File: apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx [hunk 2]
```diff
@@ -65,6 +65,23 @@ const CreateRideDialog = ({
...
+        break;
+      default:
+        break;
+    }
+  };
+  const isGasMoneyNegative = Number(gasMoney) < 0;
+  const isSeatsLessThanOne = Number(seats) < 1;
+  const isSeatsNotInteger = !Number.isInteger(Number(seats));
+
+
```
Comment (Zoeshwa) [inline apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx:84]: I recommend seeing how validations were done during the registration process or connecting to the system

File: apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx [hunk 3]
```diff
@@ -65,6 +65,23 @@ const CreateRideDialog = ({
...
   const handleSubmit = async () => {
     if (!community || !startLocation || !destination || !gasMoney || !seats) {
       alert('All fields are required.');
```
Comment (Zoeshwa) [inline apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx:87]: change the alert to write to const error and display the error in the dialog

File: apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx [hunk 4]
```diff
@@ -168,14 +188,17 @@ const CreateRideDialog = ({
...
+          helperText={isSeatsLessThanOne ? 'Seats must be 1 or more' : isSeatsNotInteger
+            ? 'Seats must be complete' : '' }
         />
       </DialogContent>
       <DialogActions>
         <Button onClick={handleClose} disabled={isLoading}>
           {TEXT.CANCEL}
         </Button>
-        <SubmitButton type="submit" disabled={isLoading} onClick={handleSubmit}>
+        <SubmitButton type="submit" disabled={isLoading || isGasMoneyNegative || isSeatsLessThanOne || isSeatsNotInteger} onClick={handleSubmit}>
```
Comment (Zoeshwa) [inline apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx:201]: export this into a const

File: apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx [hunk 5]
```diff
@@ -168,14 +188,17 @@ const CreateRideDialog = ({
           type="number"
           fullWidth
           value={seats}
-          onChange={(e) => setSeats(e.target.value)}
+          onChange={(e) => handleGasSeatsChange('seats', e.target.value)}
+          error={isSeatsLessThanOne || isSeatsNotInteger}
+          helperText={isSeatsLessThanOne ? 'Seats must be 1 or more' : isSeatsNotInteger
+            ? 'Seats must be complete' : '' }
         />
```
Comment (Zoeshwa) [inline apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx:195]: use consts instead

File: apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx [hunk 6]
```diff
@@ -150,7 +168,9 @@ const CreateRideDialog = ({
           type="number"
           fullWidth
           value={gasMoney}
-          onChange={(e) => setGasMoney(e.target.value)}
+          onChange={(e) => handleGasSeatsChange('gasMoney', e.target.value)}
+          error={isGasMoneyNegative}
+          helperText={isGasMoneyNegative ? 'Gas Money cannot be negative' : ''}
```
Comment (Zoeshwa) [inline apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx:173]: use consts instead with useState or useMemo

File: apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx (hunk 2 above)
Comment (Zoeshwa) [reply apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx:84]: In addition, consider saving the validation logic in a separate validation file (there is one somewhere) and only call the function with the parameters.
It is better for the component not to know what validation is being done, but only to perform validation

**Review by Zoeshwa — CHANGES_REQUESTED** [review]
Cool, left some comments for you :)
also you need to make sure to do validation in the update ride as well

File: apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx (hunk 1 above)
Comment (galpinto1308) [reply apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx:207]: added for update ride now

File: apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx (hunk 2 above)
Comment (galpinto1308) [reply apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx]: added external validation file for rides

File: apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx (hunk 4 above)
Comment (galpinto1308) [reply apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx]: combined them into a single const

File: apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx (hunk 5 above)
Comment (galpinto1308) [reply apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx]: simple sentence, decided to leave it here

File: apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx (hunk 6 above)
Comment (galpinto1308) [reply apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx]: simple sentence decided to leave it here

File: apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx (hunk 3 above)
Comment (galpinto1308) [reply apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx:91]: changed it to array of errors

File: apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx [hunk 7]
```diff
@@ -65,6 +71,21 @@ const CreateRideDialog = ({
     setOpen(false);
   };
 
+  const handleChange = (fieldName: string, value: string) => {
+    const error = validateField(fieldName, value);
+    setValidationErrors((prev) => ({ ...prev, [fieldName]: error ? error : null }));
+    switch (fieldName) {
```
Comment (Kovler10) [inline apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx:77]: move switch case to an object. then `fieldName` can be `fieldName: typeof keyof object` and you can use `object[fieldName]();` for better type safety

**Review by Kovler10 — CHANGES_REQUESTED** [review]

File: apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx (hunk 7 above)
Comment (galpinto1308) [reply apps/ui/src/Pages/RidesFeed/CreateRideDialog.tsx]: @Kovler10 
Changed it to an object

**Review by Kovler10 — APPROVED** [review]

**Pull request merged**